tests/api/api/test_artifacts.py

```diff
@@ -272,18 +272,27 @@ def test_list_artifact_with_multiple_tags(db: Session, client: TestClient):
             "identifiers": [(mlrun.api.schemas.ArtifactIdentifier(key=KEY).dict())],
         },
     )
-    # list all artifacts
-    resp = client.get(LIST_API_ARTIFACTS_PATH_WITH_TAG.format(project=PROJECT, tag="*"))
-    assert resp.status_code == HTTPStatus.OK.value
 
-    # expected to return three artifacts with the same key but different tags (latest, tag1, tag2)
-    artifacts = resp.json()["artifacts"]
-    assert len(artifacts) == 3
+    # ensure default flow (no tag) and flow where tag is '*' produce all tags
+    for artifacts_path in [
+        API_ARTIFACTS_PATH.format(project=PROJECT),
+        LIST_API_ARTIFACTS_PATH_WITH_TAG.format(project=PROJECT, tag="*"),
+    ]:
+
+        # list all artifacts
+        resp = client.get(artifacts_path)
+        assert resp.status_code == HTTPStatus.OK.value
+
+        # expected to return three artifacts with the same key but different tags (latest, tag1, tag2)
+        artifacts = resp.json()["artifacts"]
+        assert len(artifacts) == 3
 
-    tags = []
-    for artifact in artifacts:
-        assert artifact["metadata"]["tag"] in [tag, new_tag, "latest"]
-        tags.append(artifact["metadata"]["tag"])
+        tags = []
+        for artifact in artifacts:
+            assert artifact["metadata"]["tag"] in [tag, new_tag, "latest"]
+            tags.append(artifact["metadata"]["tag"])
```

Comment:
note that this check is a bit problematic as it could be only `latest` for example and still match the assert, but we defend it with the assert that coming afterwards

Reply:
removed. it's not needed.